numpy/ma/core.py

```diff
@@ -5635,6 +5635,18 @@ def ptp(self, axis=None, out=None, fill_value=None):
         np.subtract(out, min_value, out=out, casting='unsafe')
         return out
 
+    def partition(self, *args, **kwargs):
+        warnings.warn("Warning: 'partition' will ignore the 'mask' "
+                      "of the {}.".format(self.__class__.__name__),
+                      stacklevel=2)
+        return super(MaskedArray, self).partition(*args, **kwargs)
+
+    def argpartition(self, *args, **kwargs):
+        warnings.warn("Warning: 'argpartition' will ignore the 'mask' "
+                      "of the {}.".format(self.__class__.__name__),
+                      stacklevel=2)
```

Comment:
Python docs seemed to suggest that `stacklevel=2` is what you should use in a warning helper function. But yeah, this is better for consistency